### PR TITLE
[M] CANDLEPIN-531: Update RHEL dependency from pki-servlet-engine to tomcat

### PR DESCRIPTION
**Pull request by bakajstep**
- Change package from pki-servelet-engine to tomcat in candlepin.spec.tmpl and for cs9 images

### CODE REVIEW

**Comment by bakajstep** [comment]
I can only change the package for CS9 because CS8 still has the old name.

File: candlepin.spec.tmpl [hunk 1]
```diff
@@ -44,11 +44,7 @@ BuildRequires: selinux-policy-doc
 
 Requires: java-17 >= 1:17.0.0
 Requires: wget
-%if 0%{?fedora}
-Requires: tomcat
-%else
-Requires: pki-servlet-engine
-%endif
+Requires: tomcat < 10
```
Comment (nikosmoum) [inline candlepin.spec.tmpl:47]: After more discussion with Satellite/RHCS/tomcat folks, we decided not to make this update for RHEL8 or existing versions of Satellite. But we will make it for RHEL9.
Can we change this logic to be the following then?:
If (fedora or rhel >= 9)
  Requires: tomcat
else
  Requires: pki-servlet-engine

**Review by nikosmoum — CHANGES_REQUESTED** [review]

File: candlepin.spec.tmpl (hunk 1 above)
Comment (bakajstep) [reply candlepin.spec.tmpl:49]: done

File: candlepin.spec.tmpl [hunk 2]
```diff
@@ -44,8 +44,9 @@ BuildRequires: selinux-policy-doc
 
 Requires: java-17 >= 1:17.0.0
 Requires: wget
-%if 0%{?fedora}
-Requires: tomcat
+# TODO: remove the version restriction on Tomcat once CP supports the Java EE namespace changes
+%if (fedora or rhel >= 9)
```
Comment (ptoscano) [inline candlepin.spec.tmpl:48]:
```suggestion
%if 0%{?fedora} or 0%{?rhel} >= 9
```

Comment (bakajstep) [reply candlepin.spec.tmpl:48]: done

File: candlepin.spec.tmpl [hunk 3]
```diff
@@ -44,8 +44,9 @@ BuildRequires: selinux-policy-doc
 
 Requires: java-17 >= 1:17.0.0
 Requires: wget
-%if 0%{?fedora}
-Requires: tomcat
+# TODO: remove the version restriction on Tomcat once CP supports the Java EE namespace changes
+%if 0%{?fedora} or 0%{?rhel} >= 9
```
Comment (nikosmoum) [inline candlepin.spec.tmpl:48]: `or` is not an rpm spec operator. this should be `||`

**Review by nikosmoum — CHANGES_REQUESTED** [review]
There's a change on the 'or' operator needed, but furthermore, this needs to be tested  on RHEL8/9 (both the building of the rpm, and installing it).

File: candlepin.spec.tmpl (hunk 3 above)
Comment (bakajstep) [reply candlepin.spec.tmpl]: done

**Comment by bakajstep** [comment]
> There's a change on the 'or' operator needed, but furthermore, this needs to be tested on RHEL8/9 (both the building of the rpm, and installing it).

i just verify it that you can build it and install on RHEL8/9 using tito

**Review by nikosmoum — APPROVED** [review]

**Pull request merged**